transport/matrix/layer.py

```diff
@@ -1,51 +1,34 @@
 import os
 import sys
-from typing import Dict, Any, List
+from typing import List
 from urllib.parse import urlparse
 
 import click
 from eth_utils import to_normalized_address, decode_hex, remove_0x_prefix
-
 from raiden.api.python import RaidenAPI
 from raiden.constants import PATH_FINDING_BROADCASTING_ROOM, MONITORING_BROADCASTING_ROOM
 from raiden.exceptions import RaidenError
+from transport.matrix.transport import MatrixNode as MatrixTransportNode
+from transport.matrix.transport import MatrixLightClientNode as MatrixLightClientTransportNode
+from transport.matrix.utils import get_available_servers_from_config, server_is_available, make_client
 from raiden.settings import DEFAULT_MATRIX_KNOWN_SERVERS
 from raiden.storage import sqlite, serialize
 from raiden.transfer import views
 from raiden.utils import typing
 from raiden.utils.cli import get_matrix_servers
 from raiden.utils.signer import recover
 from transport.layer import Layer as TransportLayer
-from transport.matrix.transport import MatrixLightClientNode as MatrixLightClientTransportNode, \
-    MatrixNode as MatrixTransportNode
-from transport.matrix.utils import get_available_servers_from_config, server_is_available, make_client
 from transport.node import Node as TransportNode
 
 
-class MatrixLayer(TransportLayer):
-
-    def __init__(self, config: Dict[str, Any]):
-        from raiden.ui.app import log
-
-        self.environment_type = config["environment_type"]
-
-        if config["transport"]["matrix"].get("available_servers") is None:
-            # fetch list of known servers from raiden-network/raiden-tranport repo
-            available_servers_url = DEFAULT_MATRIX_KNOWN_SERVERS[self.environment_type]
-            available_servers = get_matrix_servers(available_servers_url)
-            log.debug("Fetching available matrix servers", available_servers=available_servers)
-            config["transport"]["matrix"]["available_servers"] = available_servers
+class MatrixLayer(TransportLayer[MatrixTransportNode]):
 
-        # TODO: This needs to be adjusted once #3735 gets implemented
-        # Add PFS broadcast room if enabled
-        if config["services"]["pathfinding_service_address"] is not None:
-            if PATH_FINDING_BROADCASTING_ROOM not in config["transport"]["matrix"]["global_rooms"]:
-                config["transport"]["matrix"]["global_rooms"].append(PATH_FINDING_BROADCASTING_ROOM)
-
-        # Add monitoring service broadcast room if enabled
-        if config["services"]["monitoring_enabled"] is True:
-            config["transport"]["matrix"]["global_rooms"].append(MONITORING_BROADCASTING_ROOM)
+    def construct_full_node(self, config):
+        self._prepare_config(config)
+        return MatrixTransportNode(config["address"], config["transport"]["matrix"])
 
+    def construct_light_clients_nodes(self, config):
+        self._prepare_config(config)
```

Comment:
i'm not sure it makes sense to call `_prepare_config` twice. what if we moved it to the constructor?

Reply:
done